### PR TITLE
feat: add account-list component

### PR DESCRIPTION
**Pull request by dcocking**
### Type of change

- [ ] Chore
- [X] Feature
- [ ] Bug fix

### Linked issue
Story: https://cybrid.atlassian.net/browse/CYB-571

Component creation: https://cybrid.atlassian.net/browse/CYB-590
Unit testing: https://cybrid.atlassian.net/browse/CYB-591

### Why
We want to display all cryptocurrency accounts and their balances.

### How
Added:
- `account-list.component`
- `account.service`
- Unit testing for both the service and component

Screen Shot 2022-08-01 at 2 18 13 PM
Screen Shot 2022-08-01 at 2 18 27 PM

### CODE REVIEW

File: library/src/app/components/account-list/account-list.component.ts [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+  }
+
+  balance$ = new BehaviorSubject<number>(0);
+
+  isLoading$ = new BehaviorSubject(true);
+  isRecoverable$ = new BehaviorSubject(true);
+  private unsubscribe$ = new Subject();
+
+  // todo(Dustin) look into getting counter_asset as part of the config
+  counter_asset = Constants.USD_ASSET;
```
Comment (brent-cybrid) [inline library/src/app/components/account-list/account-list.component.ts:49]: How are we going to handle banks that support multiple fiat assets?

Should we support a list here that is possibly empty? To handle cases where:

a) the bank does not want to show the fiat value
b) the bank wants to show the fiat value in a single fiat asset, e.g., CAD
c) the bank wants to show the fiat value in multiple fiat assets, e.g., CAD and USD

File: library/src/shared/services/account/account.service.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+  ) {}
+
+  ngOnDestroy() {
+    this.unsubscribe$.next('');
+    this.unsubscribe$.complete();
+  }
+
+  // Filter for 'trading' accounts
+  filterAccounts(): Observable<AccountBankModel[]> {
+    return this.accountsService.listAccounts().pipe(
```
Comment (brent-cybrid) [inline library/src/shared/services/account/account.service.ts:54]: Does accountsService.listAccounts() take care of pagination, i.e., will it list all the accounts regardless of how many pages of accounts there are?

File: library/src/shared/pipes/asset/asset.pipe.ts [hunk 3]
```diff
@@ -108,10 +111,7 @@ export class AssetPipe implements PipeTransform, OnDestroy {
             Constants.MIN_FRACTION_DIGITS.toString() +
             '-' +
             asset.decimals.toString();
-          return (
-            asset.symbol +
-            formatNumber(tradeUnit.toNumber(), this.locale, digitsInfo)
-          );
+          return formatNumber(tradeUnit.toNumber(), this.locale, digitsInfo);
```
Comment (brent-cybrid) [inline library/src/shared/pipes/asset/asset.pipe.ts:114]: Why did we refactor the symbol out of the pipe? Won't this cause issues with consistency across the app?

**Review by brent-cybrid — COMMENTED** [review]
As mentioned on Slack,

Looks like the display on the account listing isn't consistently following localization. Looks like the USD value in FR isn't formatted correctly

The total in FR also doesn't look correct

File: library/src/shared/pipes/asset/asset.pipe.ts (hunk 3 above)
Comment (dcocking) [reply library/src/shared/pipes/asset/asset.pipe.ts:114]: I had already refactored code across the app to reflect this change. The pipe takes a whole asset model so the calling component always has the symbol, which we can easily add to the template. I did this to try and reduce the amount of changes but it seems I need to add a case to the pipe to handle localization properly anyways.

File: library/src/app/components/account-list/account-list.component.ts (hunk 1 above)
Comment (dcocking) [reply library/src/app/components/account-list/account-list.component.ts:49]: This points towards adding a fiat param to the application config.

File: library/src/shared/pipes/asset/asset.pipe.ts (hunk 3 above)
Comment (dcocking) [reply library/src/shared/pipes/asset/asset.pipe.ts]: I've opted towards using Angular's built in currency pipe for numbers that don't need to be converted, just formatted. The changes to the asset pipe have been reverted.

File: library/src/app/components/account-list/account-list.component.ts (hunk 1 above)
Comment (brent-cybrid) [reply library/src/app/components/account-list/account-list.component.ts:49]: This should be a config to start. Defaulting to USD is fine, but let's build in as a single value (that if not set does not display).

Please coordinate with @nalancer08 and @jc-cybrid as well as @faramarz-cybrid

Comment (dcocking) [reply library/src/app/components/account-list/account-list.component.ts]: I've added it to the component config as a required param, default to 'USD'. We can discuss in the SDK meeting.

**Review by brent-cybrid — APPROVED** [review]

**Pull request merged**